src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -392,7 +393,7 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                                 // move from unwatched folder to watcher folder scenario or a move from the watcher folder out.
                                 // Check if the item exists on disk to check which it is
                                 // Don't send a new notification if we already sent one for this event.
-                                if (DoesItemExist(path, IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsFile)))
+                                if (DoesItemExist(path.ToString(), IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsFile)))
```

Comment:
Why are you converting back to string again?

Reply:
Directory.Exists and File.Exists doesnot have span overloads

Reply:
You have the original string (which you shouldn't, per the original task).  You're doing `eventPaths[i].AsSpan().ToString()` here.

If you fix the incoming string to be a span in some buffer then you wouldn't be allocating a copy of an already allocated string, but you still would be allocating. You can:

1. Write direct so that we don't allocate (i.e. write the equivalent of a Span Exists)
2. Make sure we have an issue tracking that we need a File/Directory exists that takes a Span.

Reply:
I have made the incoming string to be a span in some buffer.
> Write direct so that we don't allocate (i.e. write the equivalent of a Span Exists)

We will have to write a version of GetFullPath. i dont know if there is any way to use the existing ```RemoveRelativeSegments```  function from coreclr ? should we do this in this PR ?

>  Make sure we have an issue tracking that we need a File/Directory exists that takes a Span.

do we need to write an api proposal for this ?

Reply:
> We will have to write a version of GetFullPath.

Why? Paths don't need normalized before passing to the native APIs.